src/Storages/MergeTree/RPNBuilder.h

```diff
@@ -114,10 +265,9 @@ class RPNBuilder : WithContext
         return true;
     }
 
-    const AtomFromASTFunc & atom_from_ast;
-    Block block_with_constants;
-    RPN rpn;
+    RPNBuilderTreeContext tree_context;
+    const ExtractAtomFromTreeFunction & extract_atom_from_tree_function;
```

Comment:
Do we need this lambda function? It seems to have the same body all the time.

Reply:
Agreed. I prefer to replace this builder with some static function and provide some template lambda, because RPNBuilder is template.